haystack/nodes/search_engine/web.py

```diff
@@ -0,0 +1,133 @@
+import pydoc
+from typing import List, Dict, Any, Optional, Union, Tuple, Type, Callable
+
+from haystack import BaseComponent, MultiLabel, Document
+from haystack.nodes import PromptNode, PromptTemplate, TopPSampler
+from haystack.nodes.search_engine.base import SearchEngine
+
+
+class WebSearch(BaseComponent):
```

Comment:
The WebSearch class doesn't really now where the underlying search engine gets the results from. Lets keep in mind that in future, we might have a search engine that doesn't search the web but a different resource.